core/src/convert.rs

```diff
@@ -9,6 +13,60 @@ use alloc::{
     vec::Vec,
 };
 
+impl TryFrom<Ipld> for () {
+    type Error = Error;
+
+    fn try_from(ipld: Ipld) -> Result<Self, Self::Error> {
+        match ipld {
+            Ipld::Null => Ok(()),
+            _ => Err(TypeError {
+                expected: TypeErrorType::Null,
+                found: ipld.into(),
+            }
+            .into()),
+        }
+    }
+}
+
+macro_rules! derive_from_ipld_option {
```

Comment:
I should've spotted that earlier. What about renaming both to `…try_from…` instead of just `try`? I don't have strong opinions about it, if you prefer keeping it like that, it's OK for me.